### ISSUE TITLE
[#SomeChanges Suggestion] Fix Eclipse Lag for Moonkin

### ISSUE THREAD

**Issue by ssabbyccatt** [ClassicWoWCommunity/cata-classic-bugs#1409]
### Description
When Eclipse (Solar) or Eclipse (Lunar) activates, there is a brief window of time during which the Eclipse buff will not apply to any instant-cast spell. Because of this, boomkin are using the "fake queue" Weakaura, which forces the game to briefly lag and allows us to bypass the Eclipse Lag window.

Eclipse Lag has an upside and a downside, because it applies to both the beginning and end of every Eclipse. The downside is what is described above; the upside is that it allows us to queue an instant Starsurge as we exit Eclipse, and that Starsurge will still receive the Eclipse buff, even though it was technically cast outside of Eclipse.

Ideally, Eclipse Lag is removed all together. This may result in a minor DPS loss from no longer being able to cast an instant Starsurge as you leave Eclipse, however, it will remove the need to "fake queue" spells.

**Comment by NoCement** [comment]
Agree, super annoying. Either play with WeakAura that adds artificial delay and requires special behaviour (no button smashing) plus lags game for you or do manual delay in your rotation. Doesn't feel good at all. 

Also most people don't even know about this and just use DoT right away and think it does benefit from Eclipse.... why wouldn't it...

**Comment by ssabbyccatt** [comment]
Cata is ending, closing as this is no longer really relevant.
